docs/backends/bgapi.rst

```diff
@@ -0,0 +1,62 @@
+.. _bgapi-backend:
+
+Silicon Labs BGAPI backend
+=============
+
+The BGAPI backend of Bleak communicates with any device that implements Silicon Labs "BGAPI"
+Protocol.  Classically, this is a Silicon Labs microcontroller, attached via a serial port,
+which has been programmed with some variant of "NCP" (Network Co-Processor) firmware.
+
+This does `not` apply to devices using "RCP" (Radio Co-Processor) firmware, as those only
+expose the much lower level HCI interface.
+
+References:
+ * `AN1259: Using the v3.x Silicon Labs Bluetooth Stack in Network Co-Processor Mode <https://www.silabs.com/documents/public/application-notes/an1259-bt-ncp-mode-sdk-v3x.pdf>`_
+ * https://docs.silabs.com/bluetooth/5.0/index
+
+Requirements
+------
+This backend uses `pyBGAPI <https://pypi.org/project/pybgapi/>`_ to handle the protocol layers.
+
+
+Usage
+-----
+This backend cannot be automatically selected based on system type, so must be manually
+selected, using code such as:
+
+.. code-block:: python
+
+    async with bleak.BleakClient(
+            "11:aa:bb:cc:22:33",
+            backend=bleak.backends.bgapi.client.BleakClientBGAPI,
+            bgapi="/home/karlp/SimplicityStudio/SDKs/gecko-4.2.0/protocol/bluetooth/api/sl_bt.xapi",
+            adapter="/dev/ttyACM1",
```

Comment:
Instead of requiring this, it would be nice to use environment variables to pass this information. Then any Bleak-based app could be run using that backend without having to modify any code.

Reply:
sounds good, this seemed like a reasonable method to start, but I can do that.

Reply:
You'd still expect the user to provide the backend explicitly though right? I could modify the selector right now to auto select the backend if an enviironmen variable pointing to the bgapi file exists? is that what you mean?

Reply:
Yes, that way we can run all of the examples without having to modify the code.